examples/complete/privileged-principals.tf

```diff
@@ -0,0 +1,99 @@
+locals {
+  principal_names = [
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${join("", module.deployment_principal_label.*.id)}",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${join("", module.additional_deployment_principal_label.*.id)}"
+  ]
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_iam_policy" "deployment_iam_policy" {
+  count = var.privileged_principal_enabled ? 1 : 0
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "s3:*",
+        ]
+        Effect   = "Allow"
+        Resource = "arn:aws:s3:::${module.this.id}*"
+      },
+    ]
+  })
+}
+
+module "deployment_principal_label" {
+  count = var.privileged_principal_enabled ? 1 : 0
+
+  source  = "cloudposse/label/null"
+  version = "0.24.1"
+
+  attributes = ["deployment"]
+
+  context = module.this.context
+}
+
+resource "aws_iam_role" "deployment_iam_role" {
+  count = var.privileged_principal_enabled ? 1 : 0
+
+  name = join("", module.deployment_principal_label.*.id)
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = ""
+        Principal = {
+          Service = "ec2.amazonaws.com"
```

Comment:
why do we need to allow `ec2.amazonaws.com` to assume the role?
how's EC2 related to the bucket (and to the principals that we are adding in this PR)?

Reply:
This was just an arbitrary assume role policy. I'll think of other examples.

Reply:
@aknysh I am OK with this as an example of how you would make an EC2 Instance Role, although again this should be done with `aws_iam_policy_document` not `jsonencode`.